sdk/cosmosdb/cosmos/src/queryExecutionContext/EndpointComponent/OffsetLimitEndpointComponent.ts

```diff
@@ -22,9 +46,15 @@ export class OffsetLimitEndpointComponent implements ExecutionContext {
     const buffer: any[] = [];
     const response = await this.executionContext.fetchMore(diagnosticNode);
     mergeHeaders(aggregateHeaders, response.headers);
-    if (response === undefined || response.result === undefined) {
+    if (
+      response === undefined ||
+      response.result === undefined
+    ) {
       return { result: undefined, headers: response.headers };
     }
+    
+    const initialOffset = this.offset;
+    const initialLimit = this.limit;
 
     for (const item of response.result) {
```

Comment:
This iteration look inefficient. We can directly skip offset elements? And push required elements to buffer